tools/generator/test/CreateXCSchemesTests.swift

```diff
@@ -0,0 +1,184 @@
+        }
+        XCTAssertEqual(
+            buildAction.buildActionEntries,
+            expectedBuildActionEntries,
+            "buildActionEntries did not match for \(scheme.name)"
+        )
+        XCTAssertTrue(
+            buildAction.parallelizeBuild,
+            "parallelizeBuild was not true for \(scheme.name)"
+        )
+        XCTAssertTrue(
+            buildAction.buildImplicitDependencies,
+            "buildImplicitDependencies was not true for \(scheme.name)"
+        )
+
+        guard let testAction = scheme.testAction else {
+            XCTFail("Expected a test action for \(scheme.name)")
+            return
+        }
+        XCTAssertNil(
+            testAction.macroExpansion,
+            "macroExpansion was not nil for \(scheme.name)"
+        )
+        XCTAssertEqual(
+            testAction.buildConfiguration,
+            expectedBuildConfigurationName,
+            "the test action buildConfiguration did not match for \(scheme.name)"
+        )
+        XCTAssertEqual(
+            testAction.testables,
+            expectedTestables,
+            "testables did not match for \(scheme.name)"
+        )
+
+        guard let launchAction = scheme.launchAction else {
+            XCTFail("Expected a launch action for \(scheme.name)")
+            return
+        }
+        XCTAssertEqual(
+            launchAction.buildConfiguration,
+            expectedBuildConfigurationName,
+            "the launch action buildConfiguration did not match for \(scheme.name)"
+        )
+        XCTAssertEqual(
+            launchAction.runnable,
+            expectedBuildableProductRunnable,
+            "runnable did not match for \(scheme.name)"
+        )
+
+        guard let analyzeAction = scheme.analyzeAction else {
+            XCTFail("Expected an analyze action for \(scheme.name)")
+            return
+        }
+        XCTAssertEqual(
+            analyzeAction.buildConfiguration,
+            expectedBuildConfigurationName,
+            "the analyze action buildConfiguration did not match for \(scheme.name)"
+        )
+
+        guard let archiveAction = scheme.archiveAction else {
+            XCTFail("Expected an archive action for \(scheme.name)")
+            return
+        }
+        XCTAssertEqual(
+            archiveAction.buildConfiguration,
+            expectedBuildConfigurationName,
+            "the archive action buildConfiguration did not match for \(scheme.name)"
+        )
+        XCTAssertTrue(
+            archiveAction.revealArchiveInOrganizer,
+            "revealArchiveInOrganizer did not match for \(scheme.name)"
+        )
+    }
+
+    func test_createXCSchemes_WithNoTargets() throws {
+        let schemes = try Generator.createXCSchemes(
+            filePathResolver: filePathResolver,
+            pbxTargets: [:]
+        )
+        let expected = [XCScheme]()
+        XCTAssertEqual(schemes, expected)
+    }
+
+    func test_createXCSchemes_WithTargets() throws {
```

Comment:
Nit: can these follow camel case?
```suggestion
    func test_createXCSchemes_withTargets() throws {
```

Reply:
Done.